src/app/routes/repo.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{
+    app::auth::middleware::SessionInfo,
+    service::database::{
+        conn::DbPool,
+        models::{Challenge, Repository, User},
+    },
+    shared::errors::{CreateRepositoryError, RepositoryError},
+};
+use actix_web::{web, HttpMessage, HttpRequest, HttpResponse, Scope};
+use log::error;
+use reqwest::StatusCode;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CreateRepoResponse {
+    repo_name: String,
+    repo_url: String,
+}
+
+#[derive(Deserialize)]
+pub struct CreateRepoRequest {
+    repo_url: String,
+}
+
+pub fn init() -> Scope {
+    web::scope("/repo").route("", web::post().to(create_repo))
+}
+
+async fn create_repo(
+    req: HttpRequest,
+    body: Result<web::Json<CreateRepoRequest>, actix_web::Error>,
+    pool: web::Data<DbPool>,
+) -> Result<HttpResponse, RepositoryError> {
+    let repo_url = match body {
+        Ok(body) => body.repo_url.clone(),
+        Err(e) => {
+            error!("Error getting repository url: {}", e);
+            return Err(RepositoryError::BadRequest(
+                "Repository url is required".to_string(),
+            ));
+        }
+    };
+    if repo_url.is_empty() {
+        return Err(RepositoryError::BadRequest(
+            "Repository url is required".to_string(),
+        ));
+    }
+
+    let client = reqwest::Client::new();
+    let git_service_url = std::env::var("GIT_SERVICE_URL").expect("GIT_SERVICE_URL not set");
```

Comment:
```suggestion
       let git_service_url = std::env::var("GIT_SERVICE_URL").map_err(|_| {
        error!("GIT_SERVICE_URL environment variable not set");
        RepositoryError::InternalServerError("Server configuration error".to_string())
    })?;
```

The `expect` will cause the application to panic and crash in the event this env variable is not present which will effectively bring down the entire core server.  Instead we should handle this case gracefully.

Reply:
Shouldn't it be the desired behaviour for the application to crash if any environment variables are missing? The application won't work so don't you think it makes sense to panic and throw the error?

Reply:
How about adding a check in the entrypoint (main.rs) for the env variables needed to run the application. If any env variable is missing, we panic and don't start the server.
E.g.
```rust
async fn main() -> std::io::Result<()> {
    dotenv().ok();
    // check if any env's are not set
    if std::env::var("DATABASE_URL").is_err() {
        panic!("DATABASE_URL is not set");
    }
    if std::env::var("GIT_SERVICE_URL").is_err() {
        panic!("GIT_SERVICE_URL is not set");
    }
    env_logger::Builder::from_env(Env::default().default_filter_or("debug")).init();

    let pool = get_connection_pool();
    let manager_handle = WebSocketManagerHandle::new();

    HttpServer::new(move || {
        App::new()
            .app_data(web::Data::new(pool.clone()))
            .app_data(web::Data::new(manager_handle.clone()))
            .wrap(Logger::default())
            .service(
                web::scope("/api")
                    .wrap(AuthMiddleware)
                    .configure(routes::init),
            )
            .service(
                web::resource("/ws")
                    .wrap(AuthMiddleware)
                    .route(web::get().to(websocket_handler)),
            )
            .service(
                web::resource("/webhook").route(web::post().to(routes::webhook::handle_webhook)),
            )
    })
    .bind("127.0.0.1:4925")?
    .run()
    .await
}
```

Reply:
If that's the case then we should do the check during the startup of the app and not when the first request is made I think.